stellargraph/connector/neo4j/graph.py

```diff
@@ -138,6 +138,32 @@ def to_adjacency_matrix(self, node_ids):
     def is_directed(self):
         return self._is_directed
 
+    def cluster(self):
+        """
+        Performs Louvain community detection to cluster the graph.
+
+        Returns:
+             A list of lists, where each inner list corresponds to a cluster and
+              contains the node ids of the nodes in that cluster.
+        """
+        cluster_query = """
+            CALL gds.louvain.stream({
+                nodeQuery: 'MATCH (n) RETURN id(n) AS id',
+                relationshipQuery: 'MATCH (n)-->(m) RETURN id(n) AS source, id(m) AS target'
+            })
+            YIELD nodeId, communityId
+            RETURN gds.util.asNode(nodeId).ID AS node_id, communityId AS community_id
+        """
+        result = self.graph_db.run(cluster_query).data()
```

Comment:
Since this function does a single pass over the result (even with the `collect`/aggregating form), I think the `data()` call here is unnecessary:

```suggestion
        result = self.graph_db.run(cluster_query)
```

Or, assuming the group-by form in a separate comment works:

```suggestion
        return [row["node_ids"] for row in self.graph_db.run(cluster_query)]
```

(Plus deleting the rest of the function.)